functions/src/utils/permissions.ts

```diff
@@ -29,7 +29,86 @@ export const checkIsAdmin = async (uid: string) => {
   }
 };
 
-export const checkIsWorldOwner = async (worldId: string, uid: string) => {
+const checkIfSpaceOwner = async ({
+  spaceId,
+  uid,
+}: {
+  spaceId: string;
+  uid: string;
+}): Promise<boolean> => {
+  return await admin
+    .firestore()
+    .collection("venues")
+    .doc(spaceId)
+    .get()
+    .then(async (doc) => {
+      if (!doc.exists) {
+        throw new HttpsError("not-found", `Venue ${spaceId} does not exist`);
+      }
+      const venue = doc.data();
+
+      if (!venue) {
+        throw new HttpsError("internal", "No data returned");
+      }
+      if (venue.owners && venue.owners.includes(uid)) return true;
+
+      if (venue.parentId) {
+        const doc = await admin
+          .firestore()
+          .collection("venues")
+          .doc(venue.parentId)
+          .get();
+
+        if (!doc.exists) {
+          throw new HttpsError(
+            "not-found",
+            `Venue ${spaceId} references missing parent ${venue.parentId}`
+          );
+        }
+        const parentVenue = doc.data();
+
+        if (!parentVenue) {
+          throw new HttpsError("internal", "No data returned");
+        }
+        if (parentVenue.owners && parentVenue.owners.includes(uid)) return true;
```

Comment:
I don't think we want this logic. My understanding is that `parentId` is now mis-named and is used solely for the back button on a space. My understanding is that you're either in the `owners` field on the space or you're not a space owner.

Reply:
Removed in https://github.com/sparkletown/sparkle/pull/2820/commits/af825f995a7267c0da1cdbc23ea8630055576519